ldai/client.go

```diff
@@ -1,12 +1,139 @@
+	"github.com/launchdarkly/go-server-sdk/ldai/internal/datamodel"
+	"github.com/launchdarkly/go-server-sdk/v7/interfaces"
+)
+
 type ServerSDK interface {
+	JSONVariation(
+		key string,
+		context ldcontext.Context,
+		defaultVal ldvalue.Value,
+	) (ldvalue.Value, error)
+	Loggers() interfaces.LDLoggers
+	TrackMetric(
+		eventName string,
+		context ldcontext.Context,
+		metricValue float64,
+		data ldvalue.Value,
+	) error
 }
 
 type Client struct {
-	sdk ServerSDK
+	sdk    ServerSDK
+	logger interfaces.LDLoggers
+}
+
+const ldContextVariable = "ldctx"
+
+func New(sdk ServerSDK) (*Client, error) {
+	if sdk == nil {
+		return nil, fmt.Errorf("sdk must not be nil")
+	}
+	return &Client{
+		sdk:    sdk,
+		logger: sdk.Loggers(),
+	}, nil
+}
+
+func (c *Client) Config(
+	key string,
+	context ldcontext.Context,
+	defaultValue Config,
+	variables map[string]interface{},
+) (Config, *Tracker) {
+	variation, err := c.sdk.JSONVariation(key, context, ldvalue.Null())
+
+	defaultTracker := NewTracker(key, c.sdk, &defaultValue, context, c.logger)
+
+	if err != nil {
+		c.logger.Warnf("Error fetching JSON variation: %s", err.Error())
+		return defaultValue, defaultTracker
+	}
+
+	if variation.IsNull() {
+		c.logger.Warnf("JSON variation was null")
+		return defaultValue, defaultTracker
+	}
+
+	var parsed datamodel.Config
+	if err := json.Unmarshal([]byte(variation.JSONString()), &parsed); err != nil {
+		c.logger.Warnf("Error unmarshalling JSON variation: %s", err.Error())
+		return defaultValue, defaultTracker
+	}
+
+	mergedVariables := map[string]interface{}{
+		ldContextVariable: getAllAttributes(context),
+	}
+	for k, v := range variables {
+		if k == ldContextVariable {
+			c.logger.Warnf("AI model config variables contains 'ldctx' key, which is reserved")
+			continue
+		}
+		mergedVariables[k] = v
+	}
+
+	builder := NewConfig().
+		WithModelId(parsed.Model.Id).
+		WithProviderId(parsed.Provider.Id).
+		WithEnabled(parsed.Meta.Enabled)
+
+	for i, msg := range parsed.Messages {
+		content, err := interpolateTemplate(msg.Content, mergedVariables)
+		if err != nil {
+			c.logger.Errorf(
+				"Malformed message at index %d: %s", i, err.Error(),
+			)
+			return defaultValue, &Tracker{}
+		}
+		builder.WithMessage(content, msg.Role)
+	}
+
+	cfg := builder.Build()
+	return cfg, NewTracker(key, c.sdk, &cfg, context, c.logger)
+}
+
+func getAllAttributes(context ldcontext.Context) map[string]interface{} {
+	if !context.Multiple() {
+		return addSingleKindContextAttributes(context)
+	}
+
+	attributes := map[string]interface{}{
+		"kind": context.Kind(),
+		"key":  context.FullyQualifiedKey(),
+	}
+
+	for _, ctx := range context.GetAllIndividualContexts(nil) {
+		attributes[string(ctx.Kind())] = addSingleKindContextAttributes(ctx)
+	}
+
+	return attributes
+}
+
+func addSingleKindContextAttributes(context ldcontext.Context) map[string]interface{} {
+	attributes := map[string]interface{}{
+		"kind":      context.Kind(),
```

Comment:
Hmmm. This is a really slight nitpick, and it may never matter, but in the multi-kind case, I don't think we should include this here for consistency with other SDKs.

Reply:
I did add `kind` in .NET as well 🤷‍♂️ 

Can remove. But it seems like if you are expecting to be able to access `kind`, you should be able to - even if it's `multi`.

Reply:
I agree you should be able to access the `kind` as 'multi'. My concern is that this allows for `ldctx.org.kind`, which doesn't match our context schema.

Reply:
Got it. Fixed.